src/resources/js/event-editor.js

```diff
@@ -3,21 +3,143 @@ var tribe_events_event_editor = tribe_events_event_editor || {};
 /**
  * Implements behaviours that are specific to the event editor.
  */
-jQuery( function( $ ) {
-	var obj = tribe_events_event_editor,
-		$sticky_in_month_view_checkbox = $( 'input[name="EventShowInCalendar"]' ),
-		$featured_event_checkbox = $( 'input[name="feature_event"]' );
+( function ( $, obj ) {
+	'use strict';
+
+	obj = obj || {};
+
+	/**
+	 *	Setup our selectors.
+	 *
+	 *	@since TBD
+	 */
+	obj.selectors = {
+		$sticky_in_month_view_checkbox: 'input[name="EventShowInCalendar"]',
+		$featured_event_checkbox: 'input[name="feature_event"]',
+		organizer: {
+			area: '#event_tribe_organizer',
+			delete_button: '.tribe-delete-this',
+			add_button: '#event_tribe_organizer .tribe-add-post',
+			post_dropdown: '.linked-post-dropdown',
+		},
+	};
+
+	obj.organizer = {};
+	/**
+	 *	Controls logic for the Organizer delete button to display.
+	 *
+	 *	If more than 1 organizer exists, display the delete button.
+	 *	If only 1 organizer exists, hide the delete button.
+	 *
+	 *	@since TBD
+	 *	@return void
+	 */
+	obj.organizer.deleteButtonDisplayLogic = () => {
+
+		const $organizers = $( obj.selectors.organizer.area ).find( '.saved-linked-post' );
+
+		$( obj.selectors.organizer.area )
+			.find( obj.selectors.organizer.delete_button )
+			.each( function () {
+
+				// If you have more than 1 organizer then we display the delete button.
+				if ( $organizers.length > 1 ) {
+					$( this ).show();
+					return;
+				}
+
+				const $organizer_dropdown = $( obj.selectors.organizer.area + ' ' + obj.selectors.organizer.post_dropdown );
+
+				//If this is running, it's because we only have 1 organizer
+				if ( $organizer_dropdown.val() !== '-1' ) {
+					$( obj.selectors.organizer.add_button ).show();
+				}
+
+				$( this ).hide();
+
+			} );
+
+	};
+	/**
+	 *	Trigger events for bind events.
+	 *
+	 *	@since TBD
+	 */
+	obj.organizer.bindEvents = () => {
+
+		/**
+		 *	Logic for the organizer area -
+		 *	"Add Organizer" button should be hidden by default,
+		 *	only appearing when there is more than 1 organizer.
+		 *
+		 *	"Trash" icon / delete button should be hidden by default,
+		 *	only appearing when there is more than 1 organizer
+		 *
+		 *	or when an organizer has the value of -1.
+		 */
+		// Hide the "Add Organizer" button by default.
+		$( obj.selectors.organizer.add_button ).hide();
+		// Run our delete button logic.
+		obj.organizer.deleteButtonDisplayLogic();
+
+		$( obj.selectors.organizer.area ).on( 'change', obj.selectors.organizer.post_dropdown, function () {
+			if ( this.value !== '-1' ) {
+				$( obj.selectors.organizer.add_button ).show();
+			} else {
+				$( obj.selectors.organizer.add_button ).hide();
+			}
```

Comment:
This logic should probably be in it's own function - much like `obj.organizer.deleteButtonDisplayLogic();`